gravitee-am-identityprovider/gravitee-am-identityprovider-http/src/main/java/io/gravitee/am/identityprovider/http/authentication/HttpAuthenticationProvider.java

```diff
@@ -182,7 +121,135 @@
+                responseHandler = httpRequest.rxSendBuffer(Buffer.buffer(bodyRequest));
+            } else {
+                String contentTypeHeader = httpRequest.headers().get(HttpHeaders.CONTENT_TYPE);
+                switch (contentTypeHeader) {
+                    case(MediaType.APPLICATION_JSON):
+                        responseHandler = httpRequest.rxSendJsonObject(new JsonObject(bodyRequest));
+                        break;
+                    case(MediaType.APPLICATION_FORM_URLENCODED):
+                        Map<String, String> queryParameters = format(bodyRequest);
+                        MultiMap multiMap = MultiMap.caseInsensitiveMultiMap();
+                        multiMap.setAll(queryParameters);
+                        responseHandler = httpRequest.rxSendForm(multiMap);
+                        break;
+                    default:
+                        httpRequest.putHeader(HttpHeaders.CONTENT_LENGTH, String.valueOf(bodyRequest.length()));
+                        responseHandler = httpRequest.rxSendBuffer(Buffer.buffer(bodyRequest));
+                }
+            }
+        } else {
+            responseHandler = httpRequest.rxSend();
+        }
+        return responseHandler;
+    }
+
+    private Map<String, Object> processResponse(TemplateEngine templateEngine, List<HttpResponseErrorCondition> errorConditions, HttpResponse<Buffer> httpResponse) throws Exception {
+        String responseBody =  httpResponse.bodyAsString();
+        templateEngine.getTemplateContext().setVariable(AUTHENTICATION_RESPONSE_CONTEXT_KEY, new HttpIdentityProviderResponse(httpResponse, responseBody));
+
+        // process response
+        Exception lastException = null;
+        if (errorConditions != null) {
+            Iterator<HttpResponseErrorCondition> iter = errorConditions.iterator();
+            while (iter.hasNext() && lastException == null) {
+                HttpResponseErrorCondition errorCondition = iter.next();
+                if (templateEngine.getValue(errorCondition.getValue(), Boolean.class)) {
+                    Class<? extends Exception> clazz = (Class<? extends Exception>) Class.forName(errorCondition.getException());
+                    if (errorCondition.getMessage() != null) {
+                        String errorMessage = templateEngine.getValue(errorCondition.getMessage(), String.class);
+                        Constructor<?> constructor = clazz.getConstructor(String.class);
+                        lastException = clazz.cast(constructor.newInstance(new Object[]{errorMessage}));
+                    } else {
+                        lastException = clazz.newInstance();
+                    }
+                }
+            }
+        }
+
+        // if remote API call failed, throw exception
+        if (lastException != null) {
+            throw lastException;
+        }
+        if (responseBody == null) {
+            return null;
```

Comment:
This method is called twice and after each call an exception is thrown in case of null value. 
Why do we not thrown the  exception here?

Reply:
FP